### PR TITLE
Feature/stage mode popup queue component refactoring 2037

### PR DESCRIPTION
**Pull request by Daboo1998**

### CODE REVIEW

File: src/scenes/widgets/WidgetContainer.tsx [hunk 1]
```diff
@@ -135,7 +135,7 @@ const WidgetContainer: FC = () => {
           <ToolbarIconList>
             <ToolbarIcon
               title={
-                agoraStore.isStageMode && !stageModeStore.isOnStage
+                agoraStore.isStageMode && !agoraStageModeStore.isOnStage
                   ? 'You are in the audience, stage mode is on'
```
Comment (aminbksi) [inline src/scenes/widgets/WidgetContainer.tsx:139]: translation maybe?

File: src/scenes/widgets/WidgetContainer.tsx [hunk 2]
```diff
@@ -135,7 +135,7 @@ const WidgetContainer: FC = () => {
...
                   : userDevicesStore.cameraOff
                   ? 'Camera on'
```
Comment (aminbksi) [inline src/scenes/widgets/WidgetContainer.tsx:141]: same

File: src/scenes/widgets/WidgetContainer.tsx [hunk 3]
```diff
@@ -145,12 +145,12 @@ const WidgetContainer: FC = () => {
...
+                (agoraStore.isStageMode && !agoraStageModeStore.isOnStage)
               }
             />
             <ToolbarIcon
               title={
-                agoraStore.isStageMode && !stageModeStore.isOnStage
+                agoraStore.isStageMode && !agoraStageModeStore.isOnStage
                   ? 'You are in the audience, stage mode is on'
                   : userDevicesStore.muted
                   ? 'Unmute'
```
Comment (aminbksi) [inline src/scenes/widgets/WidgetContainer.tsx:156]: same

File: src/scenes/widgets/WidgetContainer.tsx [hunk 4]
```diff
@@ -145,12 +145,12 @@ const WidgetContainer: FC = () => {
...
                 userDevicesStore.isTogglingCamera ||
-                (agoraStore.isStageMode && !stageModeStore.isOnStage)
+                (agoraStore.isStageMode && !agoraStageModeStore.isOnStage)
               }
             />
             <ToolbarIcon
               title={
-                agoraStore.isStageMode && !stageModeStore.isOnStage
+                agoraStore.isStageMode && !agoraStageModeStore.isOnStage
                   ? 'You are in the audience, stage mode is on'
```
Comment (aminbksi) [inline src/scenes/widgets/WidgetContainer.tsx:154]: same

File: src/scenes/AppLayers.tsx [hunk 5]
```diff
@@ -30,20 +29,18 @@ const AppLayers: FC = ({children}) => {
...
+            className="main-container"
+            style={{
+              marginRight: agoraStore.hasJoined ? '90px' : undefined
+            }}
+          >
+            {children}
+          </div>
+          <Communication />
+        </main>
+        =
```
Comment (PAKovalev) [inline src/scenes/AppLayers.tsx:43]: ?

File: src/scenes/collaboration/pages/StageModePage/components/molecules/StageModePopup/StageModePopup.tsx [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+const StageModePopup: React.FC<StageModePopupPropsInterface> = ({info, canEnterStage}) => {
+  const {collaborationStore} = useStore();
+  const {stageModeStore} = collaborationStore;
+  const {removeRequestPopup, removeAwaitingPermissionPopup} = stageModeStore;
+
+  const {t} = useTranslation();
+
+  const handleAccept = (info: StageModePopupInfoInterface) => {
+    info
```
Comment (PAKovalev) [inline src/scenes/collaboration/pages/StageModePage/components/molecules/StageModePopup/StageModePopup.tsx:24]: maybe `await` ? not readable

File: src/scenes/collaboration/pages/StageModePage/components/molecules/StageModePopup/StageModePopup.tsx [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+      ?.onAccept?.()
+      .then((shouldClose) => shouldClose && info.userId && removeRequestPopup(info.userId));
+  };
+
+  const handleDecline = (info: StageModePopupInfoInterface) => {
```
Comment (PAKovalev) [inline src/scenes/collaboration/pages/StageModePage/components/molecules/StageModePopup/StageModePopup.tsx:29]: maybe `await` ? not readable

File: src/scenes/collaboration/stores/StageModeStore/StageModeStore.ts [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+const StageModeStore = types
+  .model('StageModeStore', {
+    popups: types.array(types.frozen<StageModePopupInfoInterface>()),
+
+    request: types.optional(RequestModel, {})
+  })
+  .actions((self) => ({
+    addRequestPopup: flow(function* (
+      userId: string,
+      options?: Omit<StageModePopupInfoInterface, 'type'>
```
Comment (PAKovalev) [inline src/scenes/collaboration/stores/StageModeStore/StageModeStore.ts:17]: Usually we use Omit for external interfaces / types. Create a new interface for this. Omit looks like dirty hack )

**Review by PAKovalev — CHANGES_REQUESTED** [review]

**Review by PAKovalev — APPROVED** [review]

**Pull request merged**